src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -138,6 +139,36 @@ export module DotNet {
     }
   }
 
+  /**
+   * Creates a JavaScript data reference that can be passed to .NET via interop calls.
+   *
+   * @param arrayBufferView The ArrayBufferView used to create the JavaScript stream reference.
+   * @returns The JavaScript data reference (this will be the same instance as the given object).
+   * @throws Error if the given value is not an Object or doesn't have a valid byteLength.
+   */
+  export function createJSStreamReference(arrayBufferView: ArrayBufferView | any): any {
+    // Check if this is an ArrayBufferView, and if it has a valid byteLength for transfer
+    // using a JSStreamReference.
+    if (!(arrayBufferView.buffer instanceof ArrayBuffer)) {
+      throw new Error(`Cannot create a JSStreamReference from the value '${arrayBufferView}' as it is not a valid ArrayBuffer.`);
```

Comment:
This seems a bit weird :) If you pass in an actual `ArrayBuffer` (as opposed to typed array, like `Uint8Array`), then the `.buffer` property will be undefined and it will complain that you didn't supply an `ArrayBuffer`, even though you did.

It would be nice to support both `ArrayBuffer` and typed arrays (e.g., `Uint8Array`) but if we do want to limit it to typed arrays, we should update the error message to say that.

Reply:
Thanks for pointing this out, I've updated the error message and will look into supporting general `ArrayBuffer`s separately.